File: python/day01.py
def p1():
    password = 0
    dial = 50
    with open("input/day01/in.txt") as f:
        for line in f.readlines():
            line = line.strip()
            num = int(line[1:])
            if line[0] == 'L':
                dial = (dial - num) % 100
            else:
                dial = (dial + num) % 100

            if dial == 0:
                password += 1
    print(password)

def p2():
    password = 0
    dial = 50
    with open("input/day01/in.txt") as f:
        for line in f.readlines():
            line = line.strip()
            num = int(line[1:])
            for n in range(num):
                if line[0] == 'L':
                    dial = (dial - 1) % 100
                else:
                    dial = (dial + 1) % 100
                if dial == 0:
                    password += 1
    print(password)

if __name__ == '__main__':
    p1()
    p2()

File: c/day01.c
#include <stdio.h>

void part1()
{
	FILE *fp;
	char c;
	int n;

	int dial;
	int password;

	dial = 50;
	password = 0;

	fp = fopen("input/day01/in.txt", "r");
	while (1) {
		if (fscanf(fp, "%c%d\n", &c, &n) != 2)
			break;

		if (c == 'R')
			dial = (dial + n) % 100;
		else
			dial = (dial - n) % 100;

		if (dial == 0)
			password++;
	}
	printf("%d\n", password);
}

void part2()
{
	FILE *fp;
	int i;
	char c;
	int n;

	int dial;
	int password;

	dial = 50;
	password = 0;

	fp = fopen("input/day01/in.txt", "r");
	while (1) {
		if (fscanf(fp, "%c%d\n", &c, &n) != 2)
			break;

		for (i = 0; i < n; ++i) {
			if (c == 'R')
				dial = (dial + 1) % 100;
			else
				dial = (dial - 1) % 100;

			if (dial == 0)
				password++;
		}
	}
	printf("%d\n", password);
}

int main()
{
	part1();
	part2();
}
